### ISSUE TITLE
partially labelled data

### ISSUE THREAD

**Issue by cloudhawk-yun** [yao8839836/text_gcn#22]
Hi Yao,

Quickly run this question pass by you. How do we reproduce the results in Section "Effects of the Size of Labeled Data"? Note that this is purely for discussion purpose. No other intention behind my question. :) Just to make sure we don't have misunderstanding. 

Anyway, there are two possible places according to the code. 

Case 1
=====

If we use the lines 67-68 in build_graph.py (listed below)

#partial labeled data
#train_ids = train_ids[:int(0.2 * len(train_ids))]

Code wise, it uses 20% of labelled training data, however, the rest of 80% will be excluded from 'x' because  line 224 

train_size = len(train_ids)

Indicates that train_size comes from train_ids only.

Personally I assume that it is not the right place

Case 2
=====

At line 226 in build_graph.py

real_train_size = train_size - val_size  # - int(0.5 * train_size)

If we go from this line, "y" only contains the labels from real_train_size hence later when using load_corpus(...) in utils.py, only "real_train_size" number of labelled data are loaded. 

In this way, you can achieve partially labelled data. 

Please let me know if my understanding is correct.

Many thanks in advance. 

Best,

V

**Comment by yao8839836** [comment]
@cloudhawk-yun 

Hi, thanks for the question.

In our experiment I used case 1, which uses 20% original training data (18% real train + 2% validation). I always used 10% as validation set given different training proportions.

There will be 50% original training data (40% real train + 10% validation) if we use real_train_size = train_size - val_size  - int(0.5 * train_size).

**Comment by cloudhawk-yun** [comment]
@yao8839836 

Thank you so much for the reply. Per your reply, does it imply that  "Figure 4: Test accuracy by varying training data proportions." in your paper  uses Case 1?

If yes, do you have any insight on the pros and cons if we use Case 2? Would that introduce unnecessary noise? or it would actually boost the accuracy?

Thanks again.

V

**Comment by yao8839836** [comment]
@cloudhawk-yun 

Yes. Figure 4 uses Case 1.  I think case 2 will boost the accuracy and perform better than Case 1, as it can provide more word co-occurrence information from 80% training data which are removed in Case 1.

**Comment by cloudhawk-yun** [comment]
@yao8839836 Thanks a lot for your assistance!! Really appreciated.
